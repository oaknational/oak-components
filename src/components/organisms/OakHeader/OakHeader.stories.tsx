import React from "react";
import { StoryObj, Meta } from "@storybook/react";

import { OakHeader } from "./OakHeader";

import { OakLink } from "@/components/molecules";

const meta: Meta<typeof OakHeader> = {
  //  "title" is the title of the story and where to look for compoent in the storybook
  title: "Components/organisms/OakHeader",
  component: OakHeader,
  tags: ["autodocs"],
  argTypes: {
    headingTitle: { type: "string" },
    authorName: { type: "string" },
    authorTitle: { type: "string" },
    backgroundColour: { type: "string" },
    heroImageSrc: { type: "string" },
    heroImageAlt: { type: "string" },
    imageSrc: { type: "string" },
    imageAlt: { type: "string" },
    subHeadingText: { type: "string" },
  },
  parameters: {
    controls: {
      include: [
        // include the argTypes from the storybook-helpers
        // ...Object.keys(colorArgTypes),
        // ...Object.keys(spacingArgTypes),
        // ...Object.keys(borderArgTypes),
        "type",
      ],
    },
  },
  args: {
    headingTitle: "How to plan a lesson: a helpful guide for teachers",
    backgroundColour: "bg-decorative3-main",
    //Breadcrumb component doesn't exist in the storybook/component library
    breadcrumbs: (
      <>
        <OakLink href="/home">Home</OakLink> {">"}
        <OakLink href="/lessons">Lessons</OakLink>
      </>
    ),
    blogProfileAuthor: "Rachel Storm",
    authorTitle: "Head of School Support",
    authorName: "Rachel Storm",
    subHeadingText:
      "Body 1 Our guide for teachers, whether you're in your ITT, an ECT or a teacher of many years experience looking for a fresh look on lesson planning, is place to dive into expertise from across the sector.",
    heroImageSrc: `https://${process.env.NEXT_PUBLIC_OAK_ASSETS_HOST}/${process.env.NEXT_PUBLIC_OAK_ASSETS_PATH}/v1698336490/sample.jpg`,
  },
};

export default meta;

type Story = StoryObj<typeof OakHeader>;

export const Default: Story = {
  render: (args) => <OakHeader {...args} />,
  args: {},
};
