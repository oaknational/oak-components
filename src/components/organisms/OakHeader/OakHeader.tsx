import React, { ReactElement, ReactNode } from "react";
import styled, { css } from "styled-components";

import {
  OakFlex,
  OakP,
  OakCloudinaryImage,
  OakHeading,
  OakBox,
  OakImage,
  OakMaxWidth,
} from "@/components/atoms";

export type OakHeaderProps = {
  headingTitle: string;
  heroImageSrc: string;
  authorImageSrc: string;
  authorName: string;
  authorTitle: string;
  subHeadingText: string;
  breadcrumbs?: ReactElement;
  children?: ReactNode;
};

// By adding the style css utils to this components css your component will be able to accept corresponding props and prop values.
// you can also add custom styles to the component by adding the styles to the css template literal below

const StyledOakImage = styled(OakImage)`
  width: 50px;
  height: 50px;
  margin-right: 12px;
  img {
    border-radius: 69px;
  }
`;

const OakHeaderCss = css<OakHeaderProps>``;

const UnstyledComponent = (props: OakHeaderProps) => {
  const {
    headingTitle,
    breadcrumbs,
    authorName,
    authorTitle,
    subHeadingText,
    heroImageSrc,
    authorImageSrc,
  } = props;
  return (
    <OakMaxWidth $background={"lavender"} $zIndex={"behind"}>
      <OakFlex
        $minWidth={"100%"}
        $flexDirection={["column-reverse", "column-reverse", "row"]}
        $justifyContent={"space-around"}
        $alignItems={["center", null]}
        $minHeight={"all-spacing-20"}
        data-testid="test"
      >
        <OakFlex
          $alignSelf={"flex-start"}
          $ph={"inner-padding-m"}
          $height={"100%"}
          $flexDirection={"column"}
        >
          <OakBox $display={["none", "none", "block"]} $pv={"inner-padding-xl"}>
            {breadcrumbs}
          </OakBox>
          <OakFlex
            $flexDirection={"column"}
            $maxWidth={"all-spacing-22"}
            $minHeight={"all-spacing-20"}
            $justifyContent={"center"}
          >
            <OakHeading
              tag="h2"
              $color={"grey80"}
              $font={["heading-4", "heading-3"]}
              $mb={"space-between-m"}
            >
              {headingTitle}
            </OakHeading>
            <OakFlex
              $mb={"space-between-m"}
              $flexDirection={"row"}
              $alignItems={"center"}
            >
              <StyledOakImage
                alt={`${authorName} profile image`}
                src={authorImageSrc}
                $zIndex={"in-front"}
              />
              <OakBox>
                <OakHeading tag="h3" $font={"heading-7"}>
                  {authorName}
                </OakHeading>
                <OakP>{authorTitle}</OakP>
              </OakBox>
            </OakFlex>
            <OakP $mb={["space-between-l", "space-between-none"]}>
              {subHeadingText}
            </OakP>
          </OakFlex>
        </OakFlex>

        <OakFlex
          $minWidth={["all-spacing-18", "all-spacing-21"]}
          $minHeight={["all-spacing-18", "all-spacing-21"]}
          $position={"relative"}
          $alignItems={"center"}
          $justifyContent={"center"}
        >
          <OakImage
            $transform={"rotate(-2.025deg)"}
            src={heroImageSrc}
            $minHeight={["all-spacing-16", "all-spacing-21", "all-spacing-20"]}
            $minWidth={["all-spacing-16", "all-spacing-21", "all-spacing-21"]}
            alt={`${headingTitle} hero image`}
            $zIndex={"in-front"}
          />
          <OakFlex
            $position={"absolute"}
            $top={"all-spacing-0"}
            $bottom={"all-spacing-0"}
            $width={["all-spacing-21", "all-spacing-23"]}
          >
            <OakCloudinaryImage
              cloudinaryId="https://res.cloudinary.com/oak-web-application/image/upload/v1699887218/ui-graphics/loopdown.svg"
              alt="loopdown ui-graphic"
              $position={"absolute"}
              $height={"100%"}
            />
          </OakFlex>
        </OakFlex>
        <OakFlex
          $alignSelf={"flex-start"}
          $ml={"space-between-m"}
          $pv={"inner-padding-xl"}
          $display={["block", "block", "none"]}
        >
          {breadcrumbs}
        </OakFlex>
      </OakFlex>
    </OakMaxWidth>
  );
};

/**
 *
 * Oak Heading component with hero image, can be used for blog posts, articles, and other content pages.
 *
 * This Oak component can be passed the OWA breadcrumbs component as a prop. Currently this component does not exist in the component library.
 */
export const OakHeader = styled(UnstyledComponent)`
  ${OakHeaderCss}
`;
