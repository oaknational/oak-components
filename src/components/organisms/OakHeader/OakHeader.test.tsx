import React from "react";
import "@testing-library/jest-dom";
import { create } from "react-test-renderer";

import { OakHeader, OakHeaderProps } from "./OakHeader";

import renderWithTheme from "@/test-helpers/renderWithTheme";

const oakHeaderProps: OakHeaderProps = {
  headingTitle: "How to plan a lesson: a helpful guide for teachers",
  authorTitle: "Head of School Support",
  authorName: "Rachel Storm",
  authorImageSrc: "https://via.placeholder.com/150",
  subHeadingText:
    "Body 1 Our guide for teachers, whether you're in your ITT, an ECT or a teacher of many years experience looking for a fresh look on lesson planning, is place to dive into expertise from across the sector.",
  heroImageSrc: `https://${process.env.NEXT_PUBLIC_OAK_ASSETS_HOST}/${process.env.NEXT_PUBLIC_OAK_ASSETS_PATH}/v1698336490/sample.jpg`,
};

describe("OakHeader", () => {
  it("renders", () => {
    const { getByTestId } = renderWithTheme(
      <OakHeader {...oakHeaderProps} data-testid="test" />,
    );
    expect(getByTestId("test")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const tree = create(<OakHeader {...oakHeaderProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
