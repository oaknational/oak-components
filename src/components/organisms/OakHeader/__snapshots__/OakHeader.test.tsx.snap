// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakHeader matches snapshot 1`] = `
.c0 {
  width: 100%;
  max-width: 30rem;
  padding-left: 0rem;
  padding-right: 0rem;
  margin-left: auto;
  margin-right: auto;
  background: #a0b6f2;
  font-family: Lexend,sans-serif;
  z-index: -1;
}

.c2 {
  min-width: 100%;
  min-height: 22.5rem;
  font-family: Lexend,sans-serif;
}

.c4 {
  height: 100%;
  padding-left: 1rem;
  padding-right: 1rem;
  font-family: Lexend,sans-serif;
}

.c6 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
  display: none;
  font-family: Lexend,sans-serif;
}

.c7 {
  max-width: 40rem;
  min-height: 22.5rem;
  font-family: Lexend,sans-serif;
}

.c10 {
  margin-bottom: 1.5rem;
  font-family: Lexend,sans-serif;
}

.c12 {
  position: relative;
  width: 100%;
  font-family: Lexend,sans-serif;
}

.c15 {
  font-family: Lexend,sans-serif;
}

.c19 {
  position: relative;
  min-width: 11.25rem;
  min-height: 11.25rem;
  font-family: Lexend,sans-serif;
}

.c21 {
  position: relative;
  width: 100%;
  min-width: 7.5rem;
  min-height: 7.5rem;
  -webkit-transform: rotate(-2.025deg);
  -ms-transform: rotate(-2.025deg);
  transform: rotate(-2.025deg);
  font-family: Lexend,sans-serif;
  z-index: 1;
}

.c22 {
  position: absolute;
  top: 0rem;
  bottom: 0rem;
  width: 30rem;
  font-family: Lexend,sans-serif;
}

.c24 {
  position: absolute;
  width: 100%;
  height: 100%;
  font-family: Lexend,sans-serif;
}

.c25 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
  margin-left: 1.5rem;
  display: block;
  font-family: Lexend,sans-serif;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c26 {
  display: block;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c14 {
  object-fit: contain;
}

.c9 {
  font-family: Lexend,sans-serif;
  font-weight: 600;
  font-size: 2rem;
  line-height: 2.5rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
  margin-bottom: 1.5rem;
  color: #1b1b1b;
}

.c16 {
  font-family: Lexend,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c17 {
  font-family: Lexend,sans-serif;
}

.c18 {
  font-family: Lexend,sans-serif;
  margin-bottom: 3rem;
}

.c13 {
  width: 50px;
  height: 50px;
  margin-right: 12px;
}

.c13 img {
  border-radius: 69px;
}

@media (min-width:750px) {
  .c0 {
    max-width: 80rem;
  }
}

@media (min-width:750px) {
  .c0 {
    padding-left: 0.75rem;
  }
}

@media (min-width:750px) {
  .c0 {
    padding-right: 0.75rem;
  }
}

@media (min-width:750px) {
  .c6 {
    display: none;
  }
}

@media (min-width:1280px) {
  .c6 {
    display: block;
  }
}

@media (min-width:750px) {
  .c19 {
    min-width: 30rem;
  }
}

@media (min-width:750px) {
  .c19 {
    min-height: 30rem;
  }
}

@media (min-width:750px) {
  .c21 {
    min-width: 30rem;
  }
}

@media (min-width:1280px) {
  .c21 {
    min-width: 30rem;
  }
}

@media (min-width:750px) {
  .c21 {
    min-height: 30rem;
  }
}

@media (min-width:1280px) {
  .c21 {
    min-height: 22.5rem;
  }
}

@media (min-width:750px) {
  .c22 {
    width: 60rem;
  }
}

@media (min-width:750px) {
  .c25 {
    display: block;
  }
}

@media (min-width:1280px) {
  .c25 {
    display: none;
  }
}

@media (min-width:750px) {
  .c3 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media (min-width:1280px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:750px) {

}

@media (min-width:750px) {
  .c26 {
    display: block;
  }
}

@media (min-width:1280px) {
  .c26 {
    display: none;
  }
}

@media (min-width:750px) {
  .c9 {
    font-weight: 600;
  }
}

@media (min-width:750px) {
  .c9 {
    font-size: 2.5rem;
  }
}

@media (min-width:750px) {
  .c9 {
    line-height: 3rem;
  }
}

@media (min-width:750px) {
  .c9 {
    -webkit-letter-spacing: 0.0115rem;
    -moz-letter-spacing: 0.0115rem;
    -ms-letter-spacing: 0.0115rem;
    letter-spacing: 0.0115rem;
  }
}

@media (min-width:750px) {
  .c18 {
    margin-bottom: 0rem;
  }
}

<div
  className="c0 c1"
>
  <div
    className="c2 c3"
    data-testid="test"
  >
    <div
      className="c4 c5"
    >
      <div
        className="c6"
      />
      <div
        className="c7 c8"
      >
        <h2
          className="c9"
        >
          How to plan a lesson: a helpful guide for teachers
        </h2>
        <div
          className="c10 c11"
        >
          <div
            className="c12 c13"
          >
            <img
              alt="Rachel Storm profile image"
              className="c14"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              sizes="100vw"
              src="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=3840&q=75"
              srcSet="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=3840&q=75 3840w"
              style={
                {
                  "bottom": 0,
                  "color": "transparent",
                  "height": "100%",
                  "left": 0,
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": "100%",
                }
              }
            />
          </div>
          <div
            className="c15"
          >
            <h3
              className="c16"
            >
              Rachel Storm
            </h3>
            <p
              className="c17"
            >
              Head of School Support
            </p>
          </div>
        </div>
        <p
          className="c18"
        >
          Body 1 Our guide for teachers, whether you're in your ITT, an ECT or a teacher of many years experience looking for a fresh look on lesson planning, is place to dive into expertise from across the sector.
        </p>
      </div>
    </div>
    <div
      className="c19 c20"
    >
      <div
        className="c21"
      >
        <img
          alt="How to plan a lesson: a helpful guide for teachers hero image"
          className="c14"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fmock-cloudinary-cloud%2Fimage%2Fupload%2Fv1698336490%2Fsample.jpg&w=3840&q=75"
          srcSet="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fmock-cloudinary-cloud%2Fimage%2Fupload%2Fv1698336490%2Fsample.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fmock-cloudinary-cloud%2Fimage%2Fupload%2Fv1698336490%2Fsample.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fmock-cloudinary-cloud%2Fimage%2Fupload%2Fv1698336490%2Fsample.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fmock-cloudinary-cloud%2Fimage%2Fupload%2Fv1698336490%2Fsample.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fmock-cloudinary-cloud%2Fimage%2Fupload%2Fv1698336490%2Fsample.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fmock-cloudinary-cloud%2Fimage%2Fupload%2Fv1698336490%2Fsample.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fmock-cloudinary-cloud%2Fimage%2Fupload%2Fv1698336490%2Fsample.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fmock-cloudinary-cloud%2Fimage%2Fupload%2Fv1698336490%2Fsample.jpg&w=3840&q=75 3840w"
          style={
            {
              "bottom": 0,
              "color": "transparent",
              "height": "100%",
              "left": 0,
              "objectFit": undefined,
              "objectPosition": undefined,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="c22 c23"
      >
        <div
          className="c24"
        >
          <img
            alt="loopdown ui-graphic"
            className="c14"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699887218/ui-graphics/loopdown?_a=BAVAfVAO0"
            style={
              {
                "bottom": 0,
                "color": "transparent",
                "height": "100%",
                "left": 0,
                "objectFit": undefined,
                "objectPosition": undefined,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="c25 c26"
    />
  </div>
</div>
`;
